### ISSUE TITLE
[Bug] Ember Canary breaks build

### ISSUE THREAD

**Issue by pichfl** [emberjs/ember.js#20281]
### 🐞 Describe the Bug

I'm a maintainer of [ember-promise-modals](https://github.com/mainmatter/ember-promise-modals) and all recent pipelines—even ones that do not include any code changes at all—are failing on recent Canary builds.

### 🔬 Minimal Reproduction

See any runs of our test suite: https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590

Or: 
- Install `ember-source@4.10.0-alpha.1.canary+636e3`
- Run tests locally with `CI=true` (note: they pass if `CI=false`! This nearly cost me my sanity)

### 😕 Actual Behavior

Unchanged tests do not complete.

<details>
<summary>See error log</summary>
<pre><code>
not ok 1 Chrome 107.0 - [undefined ms] - Global error: Uncaught TypeError: (0 , _emberBabel.extends) is not a function at http://localhost:7357/assets/vendor.js, line 13677
[63](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:64)
    ---
[64](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:65)
        browser log: |
[65](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:66)
            {"type":"error","text":"Uncaught TypeError: (0 , _emberBabel.extends) is not a function at http://localhost:7357/assets/vendor.js, line 13677\n","testContext":{}}
[66](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:67)
    ...
[67](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:68)
not ok 2 Chrome 107.0 - [undefined ms] - Global error: Uncaught TypeError: (0 , _emberBabel.extends) is not a function at http://localhost:7357/assets/vendor.js, line 13677
[68](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:69)
    ---
[69](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:70)
        browser log: |
[70](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:71)
            {"type":"error","text":"Uncaught TypeError: (0 , _emberBabel.extends) is not a function at http://localhost:7357/assets/vendor.js, line 13677\n","testContext":{"state":"complete"}}
[71](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:72)
    ...
[72](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:73)
not ok 3 Chrome 107.0 - [undefined ms] - Global error: Uncaught Error: Could not find module `@ember/test-helpers` imported from `dummy/tests/test-helper` at http://localhost:7357/assets/vendor.js, line 266
[73](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:74)
    ---
[74](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:75)
        browser log: |
[75](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:76)
            {"type":"error","text":"Uncaught Error: Could not find module `@ember/test-helpers` imported from `dummy/tests/test-helper` at http://localhost:7357/assets/vendor.js, line 266\n","testContext":{"state":"complete"}}
[76](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:77)
    ...
[77](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:78)
not ok 4 Chrome 107.0 - [undefined ms] - Global error: Uncaught Error: The tests file was not loaded. Make sure your tests index.html includes "assets/tests.js". at http://localhost:7357/346326029182/tests/index.html?hidepassed, line 46
[78](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:79)
    ---
[79](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:80)
        browser log: |
[80](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:81)
            {"type":"error","text":"Uncaught Error: The tests file was not loaded. Make sure your tests index.html includes \"assets/tests.js\". at http://localhost:7357/346326029182/tests/index.html?hidepassed, line 46\n","testContext":{"state":"complete"}}
[81](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:82)
    ...
[82](https://github.com/mainmatter/ember-promise-modals/actions/runs/3534915439/jobs/5932359590#step:6:83)
```
</code></pre></details>

### 🤔 Expected Behavior

Canary shouldn't fail my tests.

### 🌍 Environment

- Ember: Canary
- Node.js/npm: LTS
- OS: -
- Browser: -

### ➕ Additional Context

@mansona was mad enough to track down the trigger for this problem to this commit: https://github.com/emberjs/ember.js/commit/4e62cd659f296361ad76de47ebc063e938160db0 - I have no idea what to do here.

**Comment by chriskrycho** [comment]
Thanks for the report. @dfreeman saw something similar last week, but it resolved itself.

Bizarrely, that config change should have literally *zero* impact on the generated build, but obviously *something* is going wrong there. Possibly the issue is [this line](https://github.com/emberjs/ember.js/commit/4e62cd659f296361ad76de47ebc063e938160db0#diff-e8719582f15e67e3c95d4d92394728e9278f2da27ea6824876a73a4d0a0c5765R5), which I did not intend to introduce (I was experiment separately). It's possible something in the interaction between Ember CLI's custom handling for the pre-built Ember.js versions we distribute (we avoid re-building it for you) expects ES2017. I'll poke at that tomorrow morning and see.

Sorry for the annoyance, and thanks for running builds against Canary so that we catch these things early!

**Comment by dfreeman** [comment]
Aha, thank you for the ping @chriskrycho! I still hadn't managed to get our CI compatibility happy again and had just been putting off that work and blaming it on caching issues 😅

**Comment by chriskrycho** [comment]
Resolved by #20283. Thanks again for running against canary, folks! 🚀
